### ISSUE TITLE
Question : How is highlighting handled when building grammar bundles? Specifically on Windows? 

### ISSUE THREAD

**Issue by sortofsleepy** [emacs-tree-sitter/tree-sitter-langs#313]
👋

I've been learning a bit about Treesitter and how grammars are built the past day or so. I've managed to successfully compile a grammar library and integrate it into Emacs almost without issue...the remaining thing  I can't figure out is how to get syntax highlighting to work well.

* When I compare the library I built myself with the bundles that are being output in this repo, there's clearly a difference(see below). 
* I noticed that the native install function in Emacs appears to match with the output from my build.
* The more I research the more it's sounding like it has something to do with the queries which I guess are different in this repository? 

On a possibly related note - I tried running the Powershell script after cloning the repository but I can't get any output for some reason. Is there a step I'm missing? `tree-sitter` is currently installed via Cargo. When I run `script/compile <language>` it seems to get as far as running the tests before finishing without errors. However nothing appears to have gotten built.

Any pointing in the right direction would be greatly appreciated. At the end of the day it's not an important issue as I can just use dlls from here or use another package for syntax highlighting but, well, this has kind of grown into an itch I can't seem to stop scratching haha. 

For comparison 
===
_(Top is library from this repo, bottom is the one I built from Mingw / natively built in emacs with `treesit-install-language-grammar`)_ 
![js-highlight-pre-built](https://github.com/emacs-tree-sitter/tree-sitter-langs/assets/308302/3b403890-9ee3-41a8-ad7c-90e8448442ef)
![js-highlight-self-built](https://github.com/emacs-tree-sitter/tree-sitter-langs/assets/308302/833e72e1-d25c-4755-b815-7bd894bbe028)

**Comment by jcs090218** [comment]
Okay, there is a lot of information going on. There is some information that may help. 🤔 

1. `treesit.el` is the built-in TreeSitter module from Emacs. `tree-sitter.el` (this organization) is the external package. Therefore, they are different.
2. The highlighting in this package is easy. `M-x tree-sitter-hl-mode` to make the highlighting function.
3. The highlighting queries are stored in the [queries](https://github.com/emacs-tree-sitter/tree-sitter-langs/tree/master/queries) folder.
4. You can check the [.github/workflows/release.yml](https://github.com/emacs-tree-sitter/tree-sitter-langs/blob/master/.github/workflows/release.yml) workflow to see how our binaries are built.

**Comment by sortofsleepy** [comment]
Thank you for the quick response! 

Apologies if I wasn't clear - my question was not asking about the differences between the package `tree-sitter.el` vs Emac's built-in `treesit.el`. 

I was just trying to understand why the syntax highlighting is different between the current grammar parsers out there and the ones built by this repository / plugin. 

In the screenshots above, it shows the difference - the first one is using the parser built from this repo, the second one is showing the result of using a parser built from the current main commit of the language I was testing(javascript in this case) 

But it seems this morning I figured out why I was getting a different result; is there a particular reason you're taking specific commits for each grammar? For example I see javascript is using `fdeb68a` but the current commit of that grammar is `098bd31`. `fdeb68a` gives me MUCH better syntax highlighting compared to `098bd31` for some reason. I'm assuming something changed with the parser between that commit and now?

Lastly with the Github workflow; I did take a look and it looks like it is just calling `script/compile all` no? I did try downloading the repository and running the script but I could not generate any output for some reason. Was I missing a step? I also didn't see anything that looked like a build step for Windows in the build script(`tree-sitter-langs-build.el`, is that in another place?

Anyways, I believe my main question is basically answered so I will close this. 
Thank you for your explanation!
